library.rb

```diff
@@ -0,0 +1,174 @@
+require_relative './book'
+require_relative './person'
+require_relative './teacher'
+require_relative './student'
+require_relative './rental'
+
+class Library
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_all_books
+    if @books.empty?
+      print "Library is empty\n\n"
+      return
+    end
+    @books.each do |book|
+      print "Title: #{book.title.capitalize}, Author: #{book.author.capitalize}\n"
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      print "No person added\n\n"
+      return
+    end
+    @people.each do |person|
+      print "[#{person.class.name}] Name: #{person.name.capitalize}, ID: #{person.id}, Age: #{person.age}\n"
+    end
+  end
+
+  def list_all_rental
+    if @rentals.empty?
+      print "No rentals exist\n\n"
+      return
+    end
+    print 'ID of person: '
+    id = gets.chomp.to_i
+    person = @people[id]
+    puts person.name
+    puts 'Rentals: '
+    all_rentals = person.rentals
+    puts 'No rental found' if all_rentals.empty?
+    all_rentals.each do |rental|
+      print "Date: #{rental.date}, Book \'#{rental.book.title}\' by #{rental.book.author}\n"
+    end
+  end
+
+  def create_student(age, name)
+    print 'Classroom: '
+    classroom = gets.chomp
+    print 'Has parent permission? [y/n]: '
+    permission = ''
+    loop do
+      print "Invalid input. Enter again \n" if permission != ''
+      permission = gets.chomp
+      break if %w[y n].include?(permission)
+    end
```

Comment:
- [ ] I enterer an upcase `N` as the input for the `Has parent permission? [y/n]:` and I got this message `Invalid input. Enter again` which doesn't identify why my input was invalid. We can make the app more user-friendly by converting the user input to `downcase` even when the user enters an `upcase` input. See suggestion of how to resolve this below.

```suggestion
    loop do
      print "Invalid input. Enter again \n" if permission != ''
      permission = gets.chomp.downcase
      break if %w[y n].include?(permission)
    end
```